programs/drift/src/state/lp_pool.rs

```diff
@@ -0,0 +1,1462 @@
+pub const MIN_SWAP_FEE: i128 = 200; // 0.75% in PERCENTAGE_PRECISION
+
+pub const MIN_AUM_EXECUTION_FEE: u128 = 10_000_000_000_000;
+
+// Delay constants
+#[cfg(feature = "anchor-test")]
+pub const SETTLE_AMM_ORACLE_MAX_DELAY: u64 = 100;
+#[cfg(not(feature = "anchor-test"))]
+pub const SETTLE_AMM_ORACLE_MAX_DELAY: u64 = 10;
+pub const LP_POOL_SWAP_AUM_UPDATE_DELAY: u64 = 0;
+#[cfg(feature = "anchor-test")]
+pub const MAX_AMM_CACHE_STALENESS_FOR_TARGET_CALC: u64 = 10000u64;
+#[cfg(not(feature = "anchor-test"))]
+pub const MAX_AMM_CACHE_STALENESS_FOR_TARGET_CALC: u64 = 0u64;
+
+#[cfg(feature = "anchor-test")]
+pub const MAX_CONSTITUENT_ORACLE_SLOT_STALENESS_FOR_AUM: u64 = 10000u64;
+#[cfg(not(feature = "anchor-test"))]
+pub const MAX_CONSTITUENT_ORACLE_SLOT_STALENESS_FOR_AUM: u64 = 2u64;
+
+#[cfg(test)]
+mod tests;
+
+#[account(zero_copy(unsafe))]
+#[derive(Default, Debug)]
+#[repr(C)]
+pub struct LPPool {
+    /// name of vault, TODO: check type + size
+    pub name: [u8; 32], // 32
+    /// address of the vault.
+    pub pubkey: Pubkey, // 32, 64
+    // vault token mint
+    pub mint: Pubkey, // 32, 96
+
+    /// token_supply? to simplify NAV calculation, or load from mint account
+    /// token_total_supply: u64
+
+    /// The current number of VaultConstituents in the vault, each constituent is pda(LPPool.address, constituent_index)
+    /// which constituent is the quote, receives revenue pool distributions. (maybe this should just be implied idx 0)
+    /// pub quote_constituent_index: u16,
+
+    /// QUOTE_PRECISION: Max AUM, Prohibit minting new DLP beyond this
+    pub max_aum: u128, // 8, 136
+
+    /// QUOTE_PRECISION: AUM of the vault in USD, updated lazily
+    pub last_aum: u128, // 8, 144
+
+    /// timestamp of last AUM slot
+    pub last_aum_slot: u64, // 8, 152
+    /// timestamp of last AUM update
+    pub last_aum_ts: i64, // 8, 160
+
+    /// Oldest slot of constituent oracles
+    pub oldest_oracle_slot: u64,
+
+    /// timestamp of last vAMM revenue rebalance
+    pub last_revenue_rebalance_ts: u64, // 8, 168
+    pub revenue_rebalance_period: u64,
+
+    /// Every mint/redeem has a monotonically increasing id. This is the next id to use
+    pub next_mint_redeem_id: u64,
+
+    /// all revenue settles recieved
+    pub total_fees_received: u128, // 16, 176
+    /// all revenues paid out
+    pub total_fees_paid: u128, // 16, 192
+
+    pub cumulative_usdc_sent_to_perp_markets: u128,
+    pub cumulative_usdc_received_from_perp_markets: u128,
+
+    pub total_mint_redeem_fees_paid: i128,
+
+    pub min_mint_fee: i64,
+    pub max_mint_fee_premium: i64,
+
+    pub constituents: u16, // 2, 194
+
+    pub bump: u8,
+
+    pub usdc_consituent_index: u16,
+
+    pub gamma_execution: u8,
+    pub xi: u8,
+    pub volatility: u64,
+}
+
+impl Size for LPPool {
+    const SIZE: usize = 296;
+}
+
+impl LPPool {
+    pub fn get_nav(&self, mint: &Mint) -> Result<u128> {
```

Comment:
not used also if if quote decimals and mint decimals are same, this returns a value with 0 precision?